src/main/java/org/maxgamer/quickshop/util/MsgUtil.java

```diff
@@ -1527,4 +1516,68 @@ public static void sendMessage(@Nullable CommandSender sender, @Nullable String.
         }
     }
 
+    public static void sendMessage(@Nullable CommandSender sender, @Nullable String key, @NotNull String... args) {
+        if (sender == null) {
+            Util.debugLog("INFO: Sending message to null sender.");
+            return;
+        }
+        if (StringUtils.isEmpty(key)) {
+            Util.debugLog("INFO: Message key is empty.");
+            return;
+        }
+        String message = MsgUtil.getMessage(key, sender, args);
+        if (StringUtils.isEmpty(message)) {
+            return;
+        }
+        try {
+            plugin.getQuickChat().send(sender, message);
+        } catch (Throwable throwable) {
+            Util.debugLog("Failed to send formatted text.");
+            sender.sendMessage(message);
+        }
+    }
+
+    public static void sendMessage(@Nullable UUID uuid, @Nullable String key, @NotNull String... args) {
+        if (uuid == null) {
+            return;
+        }
+        sendMessage(Bukkit.getPlayer(uuid), key, args);
+    }
+
+    public static boolean isJson(String str) {
+        try {
+            new JsonParser().parse(str);
+            return true;
+        } catch (JsonParseException exception) {
+            return false;
+        }
+    }
+
+    @Data
+    @AllArgsConstructor
+    @Builder
+    public static class TransactionMessage {
+        @NotNull
+        private String message;
+        @Nullable
+        private String hoverItem;
+        @Nullable
+        private String hoverText;
+
+        @NotNull
+        public static TransactionMessage fromJson(String json) {
+            try {
+                if (MsgUtil.isJson(json)) {
+                    return JsonUtil.getGson().fromJson(json, TransactionMessage.class);
+                }
+            } catch (Exception ignored) {
+            }
```

Comment:
We need to compatible with old data which using ########

Reply:
I really don't want compatible old data, maybe i will purge all outdated logs from database.